#!/usr/bin/env python
PACKAGE = "franka_position_servo"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

"""
    Joint Position Controller (joint_position_controller) smoothing parameter. Increasing the value gives higher smoothing (smaller change in target position per controller update step).
"""
gen.add("position_joint_delta_filter", double_t, 0, "Filter parameter that dictates how smoothly subsequent joint position commands are smoothed. Also responsible for speed of joint position execution.", 0.1, 0.1, 1.0) # (default, min, max)
gen.add("velocity_joint_delta_filter", double_t, 0, "Filter parameter that dictates how smoothly subsequent joint velocity commands are smoothed. Also responsible for acceleration of joints.", 0.1, 0.1, 1.0) # (default, min, max)


"""
    Joint positions controller gains
"""
controller_gains = gen.add_group("Controller_Gains")

exit(gen.generate(PACKAGE, "controller_configurations", "joint_controller_params"))
